### PR TITLE
WIP Julia 1.0 support (MOI only)

### PR DESCRIPTION
**Pull request by hellemo**
I gave Julia 1.0 support a shot this weekend, with JuMP master this solves a small LP without issues, haven't tried anything else. I commented out MPB for the time being.

### CODE REVIEW

**Comment by joaquimg** [comment]
These fixes are in the other PR #25 , and they should work for all tests.
I will merge that and you can test on master.
Then, if necessary you can open a new PR :)

**Comment by hellemo** [comment]
Ah, so you beat me to it :-) The REQUIRE should have LinQuadOptInterface 0.4 I think, at least I couldn't install it as is (with JuMP#master etc).  Tests don't work on 1.0, though.

**Comment by joaquimg** [comment]
I fixed that right after, my bad!
Go for it again!

**Comment by hellemo** [comment]
Not so much left of my initial PR, but this passes all tests on 1.0 on my system. Might need some changes for 0.7/0.6 compatibility.

File: src/Xpress.jl [hunk 1]
```diff
@@ -80,7 +81,7 @@ module Xpress
     include("xprs_solve.jl")
     include("xprs_callbacks.jl")
 
-    include("XpressSolverInterface.jl")
+    # include("XpressSolverInterface.jl")
```
Comment (joaquimg) [inline src/Xpress.jl]: we cannot remove this

File: deps/build.jl [hunk 2]
```diff
@@ -1,6 +1,7 @@
 using Compat
 
 depsfile = joinpath(dirname(@__FILE__),"deps.jl")
+using Libdl
```
Comment (joaquimg) [inline deps/build.jl]: we might need a:
```
if VERSION >= v"0.7.0"
```

File: deps/build.jl [hunk 3]
```diff
@@ -9,7 +10,7 @@ end
 function write_depsfile(path)
     f = open(depsfile,"w")
     if Compat.Sys.iswindows()
-        path = replace(path, "\\", "\\\\")
+        path = replace(path, "\\" => "\\\\")
```
Comment (joaquimg) [inline deps/build.jl]: same as above

File: deps/build.jl (hunk 2 above)
Comment (joaquimg) [reply deps/build.jl]: or a `Compat.something`

File: src/Xpress.jl (hunk 1 above)
Comment (joaquimg) [reply src/Xpress.jl]: MathProgBase works with julia 0.7 and julia 1.0

**Pull request merged**